### ISSUE TITLE
Add generic arguments of methods for cache-key generation

### ISSUE THREAD

**Issue by dittodhole** [Dresel/MethodCache#12]
Example:

```
public T[] GetAll<T>();
```

Should be something like:

```
public override T[] GetAll<T>()
{
  string text = string.Format("NAMESPACE.CLASS.GetAll_{0}", new object[]
  {
    typeof (T)
  });
  // ...
}
```

**Comment by Dresel** [comment]
Good catch, if you have time you can go for a PR, otherwise I'm going to tackle this in the course of next weeks.

**Comment by Dresel** [comment]
Will be fixed in 1.5.1, please try

    Install-Package MethodCache.Fody -Version 1.5.1-alpha1
